File: backend/src/main/java/com/finance/dto/AuthDtos.java
package com.finance.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public class AuthDtos {

    public record RegisterRequest(
            @Email @NotBlank String email,
            @NotBlank @Size(min = 8, max = 100) String password,
            @NotBlank @Size(max = 100) String displayName
    ) {
    }

    public record LoginRequest(
            @Email @NotBlank String email,
            @NotBlank String password
    ) {
    }

    public record RefreshTokenRequest(
            @NotBlank String refreshToken
    ) {
    }

    public record ForgotPasswordRequest(
            @Email @NotBlank String email
    ) {
    }

    public record ResetPasswordRequest(
            @NotBlank String token,
            @NotBlank
            @Size(min = 8, max = 100)
            @Pattern(
                    regexp = "^(?=.*[A-Za-z])(?=.*\\d).+$",
                    message = "Password must contain at least one letter and one number"
            )
            String newPassword
    ) {
    }

    public record MessageResponse(
            String message
    ) {
    }

    public record AuthResponse(
            String accessToken,
            String refreshToken,
            String tokenType,
            long expiresIn,
            UserResponse user
    ) {
    }

    public record UserResponse(
            java.util.UUID id,
            String email,
            String displayName
    ) {
    }
}

File: backend-dotnet/FinanceTracker.Api/Contracts/AuthDtos.cs
using System.ComponentModel.DataAnnotations;
using FinanceTracker.Api.Infrastructure;

namespace FinanceTracker.Api.Contracts;

public sealed record RegisterRequest(
    [param: EmailAddress(ErrorMessage = "must be a well-formed email address")]
    [param: NotBlank(ErrorMessage = "must not be blank")]
    string? email,
    [param: NotBlank(ErrorMessage = "must not be blank")]
    [param: StringLength(100, MinimumLength = 8, ErrorMessage = "size must be between 8 and 100")]
    string? password,
    [param: NotBlank(ErrorMessage = "must not be blank")]
    [param: StringLength(100, ErrorMessage = "size must be between 0 and 100")]
    string? displayName
);

public sealed record LoginRequest(
    [param: EmailAddress(ErrorMessage = "must be a well-formed email address")]
    [param: NotBlank(ErrorMessage = "must not be blank")]
    string? email,
    [param: NotBlank(ErrorMessage = "must not be blank")]
    string? password
);

public sealed record RefreshTokenRequest(
    [param: NotBlank(ErrorMessage = "must not be blank")]
    string? refreshToken
);

public sealed record ForgotPasswordRequest(
    [param: EmailAddress(ErrorMessage = "must be a well-formed email address")]
    [param: NotBlank(ErrorMessage = "must not be blank")]
    string? email
);

public sealed record ResetPasswordRequest(
    [param: NotBlank(ErrorMessage = "must not be blank")]
    string? token,
    [param: NotBlank(ErrorMessage = "must not be blank")]
    [param: StringLength(100, MinimumLength = 8, ErrorMessage = "size must be between 8 and 100")]
    [param: RegularExpression("^(?=.*[A-Za-z])(?=.*\\d).+$", ErrorMessage = "Password must contain at least one letter and one number")]
    string? newPassword
);

public sealed record MessageResponse(string message);

public sealed record AuthResponse(
    string accessToken,
    string refreshToken,
    string tokenType,
    long expiresIn,
    UserResponse user
);

public sealed record UserResponse(
    Guid id,
    string email,
    string displayName
);
